modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-fragment/itemActions.spec.ts

```diff
@@ -480,6 +480,159 @@ test.describe('Item Actions in Data Set fragment', () => {
 		});
 	});
 
+	test('Async and Headless Item Actions (multiple actions) performs UPDATE operations on items', async ({
+		dataSetManagerApiHelpers,
+		fdsFragmentPage,
+		layout,
+		page,
+	}) => {
+		const ASYNC_ITEM_ACTION_NAME = 'Async item action';
+		const ASYNC_ITEM_ACTION_URL = '/o/data-set-manager/table-sections/{id}';
+		const ASYNC_ITEM_NEW_LABEL = getRandomString();
+		const HEADLESS_ITEM_ACTION_NAME = 'Headless item action';
+		const HEADLESS_ITEM_ACTION_PERMISSION_KEY = 'update';
+		const HEADLESS_ITEM_NEW_LABEL = getRandomString();
+
+		await test.step('Create Item Actions', async () => {
+			await dataSetManagerApiHelpers.createDataSetItemAction({
+				dataSetERC,
+				label_i18n: {en_US: HEADLESS_ITEM_ACTION_NAME},
+				permissionKey: HEADLESS_ITEM_ACTION_PERMISSION_KEY,
+				requestBody: `{"name": "${HEADLESS_ITEM_NEW_LABEL}"}`,
+				type: EItemActionType.HEADLESS,
+			});
+
+			await dataSetManagerApiHelpers.createDataSetItemAction({
+				dataSetERC,
+				label_i18n: {en_US: ASYNC_ITEM_ACTION_NAME},
+				method: EAsyncActionMethod.PATCH,
+				requestBody: `{"name": "${ASYNC_ITEM_NEW_LABEL}"}`,
+				type: EItemActionType.ASYNC,
+				url: ASYNC_ITEM_ACTION_URL,
+			});
+		});
+
+		await test.step('Configure Data Set in the page', async () => {
+			await fdsFragmentPage.configureDataSetFragment({
+				dataSetLabel,
+				layout,
+			});
+		});
+
+		const datasetRow =
+			await test.step('Checkt that the Item Actions dropdown (only 2 items) is present in table row', async () => {
+				const tableRow = await page
+					.locator('.dnd-td.item-actions')
+					.first();
+
+				await expect(
+					tableRow.getByRole('button', {
+						exact: true,
+						name: 'Actions',
+					})
+				).toBeVisible;
+
+				const button = await tableRow.getByRole('button', {
+					exact: true,
+					name: 'Actions',
+				});
+				const dropdownId = await button.evaluate((node) =>
+					node.getAttribute('aria-controls')
+				);
```

Comment:
Do we need to `evaluate` here? I see we use it a lot elsewhere in tests, so it might be needed. Potentially changing this everywhere is outside of scope of this task.

Reply:
I would say that it won't be needed in 100% of the cases but it is safer to use it. Why? To get the relation between a control - button - and the content related we rely on the `aria-controls` attribute that has the `id`. The problem is that the `aria-controls` does not behave always the same in our components. Sometimes the `aria-controls` is populated when the control - button - is rendered, but sometimes we need to interact with the control before getting the attribute.
x.e. in the FDS, the button that shows the actions dropdown usually has the attribute populated after render. However, in the Pagination, the button that shows the item selector (4 items, 8 items, ...) adds the attribute after clicking on it.

So, the idea is to ensure that we don't receive `undefined` for those components.

Reply:
Why does `evaluate` help in that case? :thinking: Does it add some delay until the attribute is available?

Reply:
AFAIK, locators do not expose DOM attributes and we need them to get the reference to the related element.